critwall/src/test/java/org/opencadc/critwall/BucketSelectorTest.java

```diff
@@ -0,0 +1,174 @@
+ *  Council of Canada nor the            Recherches du Canada ni les noms
+ *  names of its contributors may        de ses  participants ne peuvent
+ *  be used to endorse or promote        être utilisés pour approuver ou
+ *  products derived from this           promouvoir les produits dérivés
+ *  software without specific prior      de ce logiciel sans autorisation
+ *  written permission.                  préalable et particulière
+ *                                       par écrit.
+ *
+ *  This file is part of the             Ce fichier fait partie du projet
+ *  OpenCADC project.                    OpenCADC.
+ *
+ *  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+ *  you can redistribute it and/or       vous pouvez le redistribuer ou le
+ *  modify it under the terms of         modifier suivant les termes de
+ *  the GNU Affero General Public        la “GNU Affero General Public
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ ************************************************************************
+ */
+package org.opencadc.critwall;
+
+import ca.nrc.cadc.util.Log4jInit;
+import java.util.Iterator;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Test bucket selector
+ *
+ * @author jeevesh
+ */
+public class BucketSelectorTest {
+
+    private static final Logger log = Logger.getLogger(BucketSelectorTest.class);
+
+    static {
+        Log4jInit.setLevel("org.opencadc.critwall", Level.DEBUG);
+    }
+
+    public BucketSelectorTest() { }
+
```

Comment:
For all of these tests, I would add more tests to verify whitespace handling: space before, space between, spaces after. Can be multiple checks in sequence in testValidRange and testValidSingleSelector.

There seems to be code to handle config like " - 2" and "2 - " but no tests with such values. testRangeSyntax?

In general, in a test for an invalid input, make nothing but the thing that's wrong and being tested is there to obscure the intent. For example, line 133 has "5- jj" which has 3 things that catch the eye:
- asymmetric spacing (harmless but distracting)
- second value is too long (cause fail)
- second value has invalid characters (cause fail)
So - this has two things wrong in a single test so I can't tell which code path was tested: should be two separate tests with one thing wrong each: testInvalidCharacters? testInvalidBucketLength?